pkg/collector/listeners/kubelet.go

```diff
@@ -0,0 +1,217 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2017 Datadog, Inc.
+
+// +build kubelet
+
+package listeners
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	"github.com/DataDog/datadog-agent/pkg/util/docker"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/kubelet"
+	log "github.com/cihub/seelog"
+)
+
+// KubeletListener listen to kubelet pod creation
+type KubeletListener struct {
+	watcher    *kubelet.PodWatcher
+	services   map[ID]Service
+	newService chan<- Service
+	delService chan<- Service
+	ticker     *time.Ticker
+	stop       chan bool
+	m          sync.RWMutex
+}
+
+// PodContainerService implements and store results from the Service interface for the Kubelet listener
+type PodContainerService struct {
+	ID            ID
+	ADIdentifiers []string
+	Hosts         map[string]string
+	Ports         []int
+	Pid           int
+}
+
+func init() {
+	Register("kubelet", NewKubeletListener)
+}
+
+func NewKubeletListener() (ServiceListener, error) {
+	watcher, err := kubelet.NewPodWatcher(5 * time.Second)
+	if err != nil {
+		return nil, fmt.Errorf("failed to connect to kubelet, Kubernetes listener will not work: %s", err)
+	}
+	return &KubeletListener{
+		watcher:  watcher,
+		services: make(map[ID]Service),
+		ticker:   time.NewTicker(5 * time.Second),
```

Comment:
5 seconds seems overkill, I'd default to 15 seconds instead, but we will probably make that configurable later